File: Day2/src/Employee.java
public class Employee {
	int empId;
	String empName;
	float basicSalary;
	float hra;
	float pf;
	float pt;
	float medicalAllowances;
	float netSalary;
	
	public Employee() {
		System.out.println("Constructor Instantiated...");
	}
	
	Employee(int empId,String empName,int basicSalary,float medicalAllowances){
		
		this();
		
		this.empId=empId;
		this.empName=empName;
		this.basicSalary=basicSalary;
		this.hra=this.basicSalary*0.5f;
		this.pf=this.basicSalary*0.12f;
		this.pt=200;
		
	}
	
	public void getEmployeeDetails() {
		System.out.println("Emplaoyee Id : "+this.empId);
		System.out.println("Emplaoyee Name: "+this.empName);
		System.out.println("Basic Salary \t : "+this.basicSalary);
		System.out.println("HRA  : "+this.hra);
		System.out.println("PF  : "+this.pf);
		System.out.println("PT  : "+this.pt);
		System.out.println("Gross Salary  : "+this.calculateGrossSalary());
		System.out.println("Net Salary  : "+this.calculateNetSalary());
	}
	
	public float calculateGrossSalary() {
		float grossSalary=0;
		grossSalary=this.basicSalary+this.hra+this.medicalAllowances;
		return grossSalary;
	}
	
	public float calculateNetSalary() {
		float netSalary=0;
		netSalary=this.calculateGrossSalary()-this.pf+this.pt;
		return netSalary;
	}

	@Override
	public String toString() {
		return "Employee [empId=" + empId + ", empName=" + empName + ", basicSalary=" + basicSalary + ", hra=" + hra
				+ ", pf=" + pf + ", pt=" + pt + ", medicalAllowances=" + medicalAllowances + ", netSalary=" + netSalary
				+ "]";
	}
	
	
	
}

File: Day23/TypeScript-OOP/Employee.js
"use strict";
exports.__esModule = true;
var Employee = /** @class */ (function () {
    function Employee(empid, ename, salary, medical) {
        if (empid === void 0) { empid = 0; }
        if (ename === void 0) { ename = ""; }
        if (salary === void 0) { salary = 0; }
        if (medical === void 0) { medical = 0; }
        this.empid = empid;
        this.ename = ename;
        this.salary = salary;
        this.medical = medical;
        this.HRA = 0.5 * this.salary;
        this.PF = 0.12 * this.salary;
        this.PT = 200;
    }
    Employee.prototype.calculateGrossSalary = function () {
        var val = this.salary + this.HRA + this.medical;
        return val;
    };
    Employee.prototype.calculateNetSalary = function () {
        var val = this.calculateGrossSalary() - (this.PT + this.PF);
        return val;
    };
    Employee.prototype.printEmployeeDetails = function () {
        var val = {
            "empid": this.empid,
            "ename": this.ename,
            "salary": this.salary,
            "grossSalary": this.calculateGrossSalary(),
            "netSalary": this.calculateNetSalary()
        };
        return val;
    };
    return Employee;
}());
exports.Employee = Employee;
//var e = new Employee(1, "Parth", 20000, 1000);
//console.log(e.printEmployeeDetails());
